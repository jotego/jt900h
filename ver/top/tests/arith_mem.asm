    ; Arithmetic operations that read & write to memory
    main section code
    org 0
    ld a,0xbf    ; common header

    lda xix,data
    ld a,1
    add (xix),a
    ld b,(xix+)
    cp b,0xff
    jp ne,bad_end

    ld c,(xix)
    add c,0x10
    cp c,0xda
    jp ne,bad_end

end_loop:
    ld hl,0xbabe
    ld (0xffff),0xff
    jp end_loop
bad_end:
    ld hl,0xdead
    jp bad_end
data:
    dw 0xcafe,0xbeef,0xffff,0xeeee,0xcccc
    end