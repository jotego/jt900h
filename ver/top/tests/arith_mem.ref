00000701 - 0000bf18 - 00000000 - 00000000
0000ffda - 0000bf18 - 00000000 - 00000000
00000700 - 00000000 - 00000000 - 00000000
00000000 - 0000babe - 00000000 - 00000000
-- index registers --
00000071 - 00000000 - 00000000 - 00000100
SR = f942
