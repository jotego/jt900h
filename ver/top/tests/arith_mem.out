00000001 - 00000000 - 00000000 - 00000000
0000ffda - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
0000babe - 00000000 - 00000000 - 00000000
-- index registers --
00000037 - 00000000 - 00000000 - 00000000
